File: back-end/c#/TaskManagement/BLL/CustomerService.cs
using BOL;
using DAL;
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;

namespace BLL
{
    public class CustomerService
    {
        public static List<Customer> GetAllCustomers()
        {
            try
            {
                string query = "SELECT * FROM task_management.customer ORDER BY customer_name;";
                Func<MySqlDataReader, List<Customer>> func = (reader) =>
                {
                    List<Customer> customers = new List<Customer>();
                    while (reader.Read())
                    {
                        customers.Add(BaseService.InitCustomer(reader));
                    }
                    return customers;
                };

                List<Customer> customerList = DBAccess.RunReader(query, func);
                return customerList;
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
    }
}

File: back-end/php/task_management/bll/customer_service.php
<?php

class customer_service extends base_service{

    function get_all_customers() {
        $query = "SELECT * FROM task_management.customer;";
        $customers = db_access:: run_reader($query, function ($model) {
                    return $this->init_customer($model);
                });
        return $customers;
    }

}
